client-spark/spark2/src/main/java/org/apache/spark/shuffle/DelegationRssShuffleManager.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Tencent is pleased to support the open source community by making
+ * Firestorm-Spark remote shuffle server available.
+ *
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.spark.shuffle;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.ShuffleDependency;
+import org.apache.spark.SparkConf;
+import org.apache.spark.TaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.tencent.rss.client.api.CoordinatorClient;
+import com.tencent.rss.client.request.RssAccessClusterRequest;
+import com.tencent.rss.client.response.ResponseStatusCode;
+import com.tencent.rss.client.response.RssAccessClusterResponse;
+import com.tencent.rss.common.exception.RssException;
+import com.tencent.rss.common.util.Constants;
+
+public class DelegationRssShuffleManager implements ShuffleManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DelegationRssShuffleManager.class);
+
+  private final ShuffleManager delegate;
+  private final String sortShuffleManagerName;
+  private final List<CoordinatorClient> coordinatorClients;
+  private final int accessTimeoutMs;
+  private final SparkConf sparkConf;
+
+  public DelegationRssShuffleManager(SparkConf sparkConf, boolean isDriver) throws Exception {
+    this.sparkConf = sparkConf;
+    sortShuffleManagerName = sparkConf.get(
+        RssClientConfig.RSS_SORT_SHUFFLE_MANAGER_IMPL,
+        RssClientConfig.RSS_SORT_SHUFFLE_MANAGER_IMPL_DEFAULT_VALUE);
+    accessTimeoutMs = sparkConf.getInt(
+        RssClientConfig.RSS_ACCESS_TIMEOUT_MS,
+        RssClientConfig.RSS_ACCESS_TIMEOUT_MS_DEFAULT_VALUE);
+    if (isDriver) {
+      coordinatorClients = RssShuffleUtils.createCoordinatorClients(sparkConf);
```

Comment:
if shuffle with RssShuffleManager, DelegationRssShuffleManager & RssShuffleManager will have different coordinator clients. These clients should be shared.

Reply:
Add a set api in RssShuffleManager and inject the coordinator clients?

Reply:
As discussed offline, will add refactor the coordinator clients creation in next pr.